runestone/server/componentdb.py

```diff
@@ -66,20 +66,19 @@ def addQuestionToDB(self):
         srcpath, line = self.state_machine.get_source_and_line()
         subchapter = os.path.basename(srcpath).replace('.rst','')
         chapter = srcpath.split(os.path.sep)[-2]
+        grading_type = self.options.get('autograde', None)
+        print (self.arguments[0], grading_type)
 
         sel = select([questions]).where(and_(questions.c.name == self.arguments[0],
                                               questions.c.base_course == basecourse))
         res = engine.execute(sel).first()
         try:
             if res:
-                if res['question'] != self.block_text:
-                    stmt = questions.update().where(questions.c.id == res['id']).values(question = self.block_text.encode('utf8'), timestamp=last_changed)
-                    engine.execute(stmt)
+                stmt = questions.update().where(questions.c.id == res['id']).values(question = self.block_text.encode('utf8'), timestamp=last_changed, is_private='F',
+question_type=self.name, subchapter=subchapter, grading_type = grading_type, author=author,difficulty=difficulty,chapter=chapter)
```

Comment:
The metadata can change, i.e. options on the directive. And some of those
are stored in the db.

sent from mobile-- strange words are probably auto-correct gone awry.

On Aug 28, 2016 1:27 PM, "Bradley Miller" <notifications@github.com> wrote:

> In runestone/server/componentdb.py
> <https://github.com/RunestoneInteractive/RunestoneComponents/pull/250#discussion_r76535975>
> :
>
> >
> >          sel = select([questions]).where(and_(questions.c.name == self.arguments[0],
> >                                                questions.c.base_course == basecourse))
> >          res = engine.execute(sel).first()
> >          try:
> >              if res:
> > -                if res['question'] != self.block_text:
> > -                    stmt = questions.update().where(questions.c.id == res['id']).values(question = self.block_text.encode('utf8'), timestamp=last_changed)
> > -                    engine.execute(stmt)
> > +                stmt = questions.update().where(questions.c.id == res['id']).values(question = self.block_text.encode('utf8'), timestamp=last_changed, is_private='F',
> > +question_type=self.name, subchapter=subchapter, grading_type = grading_type, author=author,difficulty=difficulty,chapter=chapter)
>
> If the source code for a directive has not changed there is no reason to
> update the database.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RunestoneInteractive/RunestoneComponents/pull/250/files/7024b1d24d86aaa949fe9f5df87106fdca3aa262#r76535975>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABBvzghK0EBJTERopbBEHxiG9M01ZNdZks5qkcUEgaJpZM4JuZsm>
> .
>